### PR TITLE
[Merged by Bors] - feat: port Topology.MetricSpace.Contracting

### PR DESCRIPTION
**Pull request by int-y1**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by int-y1** [comment]
todo:

* [ ] `efixedPoint_eq_of_edist_lt_top` `efixedPoint_eq_of_edist_lt_top'` error with `symm`
* [ ] `exists_fixedPoint'` error with `rw [MapsTo.val_restrict_apply]`

**Comment by mo271** [comment]
Is this now ready for review, @int-y1?

**Comment by semorrison** [comment]
@Parcly-Taxel, it seems unhelpful to change labels without giving the author a chance to do so. What if there is still some weird gotcha the author know about and still wants to fix / comment on? Adding `awaiting-review` immediately just adds to the probability of a reviewer either wasting work rediscovering it, or missing it.

File: Mathlib/Topology/MetricSpace/Contracting.lean [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+    (hy : IsFixedPt g y) {C} (hfg : ∀ z, dist (f z) (g z) ≤ C) : dist x y ≤ C / (1 - K) :=
+  calc
+    dist x y = dist y x := dist_comm x y
+    _ ≤ dist y (f y) / (1 - K) := (hf.dist_le_of_fixedPoint y hx)
+    _ = dist (f y) (g y) / (1 - K) := by rw [hy.eq, dist_comm]
+    _ ≤ C / (1 - K) := (div_le_div_right hf.one_sub_K_pos).2 (hfg y)
+#align contracting_with.dist_fixed_point_fixed_point_of_dist_le' ContractingWith.dist_fixedPoint_fixedPoint_of_dist_le'
+
+--noncomputable section -- Porting note: Commented out, to fix an error with `end ContractingWith`.
+
```
Comment (semorrison) [inline Mathlib/Topology/MetricSpace/Contracting.lean:308]:
```suggestion
```

File: Mathlib/Topology/MetricSpace/Contracting.lean [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+variable [Nonempty α] [CompleteSpace α]
+
+variable (f)
+
+-- Porting note: Marked `fixedPoint` as noncomputable.
```
Comment (semorrison) [inline Mathlib/Topology/MetricSpace/Contracting.lean:313]:
```suggestion
```
I think it's fine to just make these changes without leaving porting notes.

**Comment by int-y1** [comment]
This is ready for review.

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**